lib/stores/localstore.ts

```diff
@@ -0,0 +1,391 @@
+import {
+  DurablePromise,
+  PendingPromise,
+  ResolvedPromise,
+  RejectedPromise,
+  CanceledPromise,
+  TimedoutPromise,
+  isPendingPromise,
+  isResolvedPromise,
+  isRejectedPromise,
+  isCanceledPromise,
+} from "../promise";
+import { IPromiseStore } from "../store";
+import { IEncoder } from "../encoder";
+import { Base64Encoder } from "../encoders/base64";
+import { ErrorCodes, ResonateError } from "../error";
+
+export class LocalPromiseStore implements IPromiseStore {
+  private readonly dbName = "resonateDB";
+  private readonly storeName = "promises";
+  private db: IDBDatabase | null = null;
+
+  constructor(private encoder: IEncoder<string, string> = new Base64Encoder()) {
+    this.initializeDb();
+  }
+
+  private async initializeDb() {
+    return new Promise<void>((resolve, reject) => {
+      const request = indexedDB.open(this.dbName, 1);
+
+      request.onerror = () => {
+        reject(request.error);
+      };
+
+      request.onsuccess = () => {
+        this.db = request.result;
+        resolve();
+      };
+
+      request.onupgradeneeded = (event) => {
+        try {
+          const db = (event.target as IDBOpenDBRequest).result;
+          db.createObjectStore(this.storeName, { keyPath: "id" });
+          resolve();
+        } catch (e: unknown) {
+          reject(e);
+        }
+      };
+    });
+  }
+
+  private async getObjectStore(): Promise<IDBObjectStore> {
+    if (!this.db) {
+      throw new ResonateError("Database not initialized", ErrorCodes.DATABASE);
+    }
+
+    const transaction = this.db.transaction(this.storeName, "readwrite");
+    return transaction.objectStore(this.storeName);
+  }
+
+  async create(
+    id: string,
+    ikey: string | undefined,
+    strict: boolean,
+    headers: Record<string, string> | undefined,
+    data: string | undefined,
+    timeout: number,
+    tags: Record<string, string> | undefined,
+  ): Promise<PendingPromise | ResolvedPromise | RejectedPromise | CanceledPromise | TimedoutPromise> {
+    const tick = Date.now();
+    this.transition(tick);
+
+    const transaction = this.db?.transaction(this.storeName, "readwrite");
```

Comment:
Is this transaction necessary? It looks like `getObjectStore` implicitly creates a transaction.